pkg/network/network.go

```diff
@@ -101,14 +101,12 @@ func Render(conf *opv1alpha1.NetworkAddonsConfigSpec, manifestDir string, opensh
 	o := []*unstructured.Unstructured{}
 	var err error
 
-	if conf.KubeMacPool != nil {
-		// render kubeMacPool
-		o, err = renderKubeMacPool(conf, manifestDir)
-		if err != nil {
-			return nil, err
-		}
-		objs = append(objs, o...)
+	// render kubeMacPool
```

Comment:
The `if` should stay here and be dropped from inside of the renderKubeMacPool function.

Reply:
the if was there in the first place, and if its there, it saves a boolean expression on both render and renderDestroy
but up to you, if you want i will take it out, wanted to strive to minimal changes + little sanity checks in the leaf user

Reply:
I value the semantics more than 3 saved lines. Please keep it outside.

Reply:
Oh, you say it was there originally. Lemme see.

Reply:
You are right. Sorry, then keep it in the original place. The single commit diff confused me.